app/models/concerns/stock_settings_override_validation.rb

```diff
@@ -46,7 +46,9 @@ def disallow_count_on_hand_if_on_demand
   end
 
   def require_count_on_hand_if_limited_stock
-    return unless on_demand == false && count_on_hand.blank?
+    return if on_demand == true || (count_on_hand.present? && 
+                                    count_on_hand.is_a?(Numeric) && 
+                                    count_on_hand.positive?)
```

Comment:
Is it correct that "On Hand" must always be a positive number (when item is not "on demand")? Also please advise if there is a more concise way of testing for this.

Reply:
I think on hand being 0 is a legit scenario and we should account for that.